src/sentry/snuba/subscriptions.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import absolute_import
+
+import json
+
+from django.db import transaction
+
+from sentry.api.event_search import get_snuba_query_args
+from sentry.snuba.models import QueryAggregations, QueryDatasets, QuerySubscription
+from sentry.utils.snuba import _snuba_pool, SnubaError
+
+query_aggregation_to_snuba = {
+    QueryAggregations.TOTAL: ("count()", "", "count"),
+    QueryAggregations.UNIQUE_USERS: ("uniq", "tags[sentry:user]", "unique_users"),
+}
+
+
+def create_snuba_subscription(
+    project, subscription_type, dataset, query, aggregation, time_window, resolution
+):
+    """
+    Creates a subscription to a snuba query.
+
+    :param project: The project we're applying the query to
+    :param subscription_type: Text identifier for the subscription type this is. Used
+    to identify the registered callback associated with this subscription.
+    :param dataset: The snuba dataset to query and aggregate over
+    :param query: An event search query that we can parse and convert into a
+    set of Snuba conditions
+    :param aggregation: An aggregation to calculate over the time window
+    :param time_window: The time window to aggregate over
+    :param resolution: How often to receive updates/bucket size
+    :return: The QuerySubscription representing the subscription
+    """
+    # TODO: Move this call to snuba into a task. This lets us successfully create a
+    # subscription in postgres and rollback as needed without having to create/delete
+    # from Snuba
```

Comment:
If this request was done in a task would you still be able to make the QuerySubscription object?

Reply:
Yeah, although I'll need to make the subscription id column nullable.